### PR TITLE
Add expiration date to SSL certificates loaded via API request

### LINKED ISSUE #723: [Improvement] Add exipration date to certificates added from API
I use a custom script to deploy wildcard SSL certificates created with letsencrypt with DNS challenge, the simple way to know if a cert is changed is using the expiration date, so I add this info to Certificates.add and Certificates.update API.

Here's the diff, is so small that I think a PR is not needed:

```
# diff ./lib/Froxlor/Api/CommandsCertificates.php.orig ./lib/Froxlor/Api/CommandsCertificates.php
295a296
>               $expirationdate = NULL;
335a337
>                       $expirationdate = $cert_content['validTo_time_t'];
349c351,352
<                       `ssl_cert_chainfile` = :ssl_cert_chainfile
---
>                       `ssl_cert_chainfile` = :ssl_cert_chainfile,
>                       `expirationdate` = :expirationdate
356a360
>                       "expirationdate" => date("Y-m-d H:i:s", $expirationdate),

```

Thanks! :)

### PR DESCRIPTION
**Pull request by kionez**
Fixes #723

### CODE REVIEW

File: lib/Froxlor/Api/Commands/Certificates.php [hunk 1]
```diff
@@ -292,6 +293,7 @@ private function addOrUpdateCertificate($domainid = 0, $ssl_cert_file = '', $ssl
 		}
 
 		$do_verify = true;
+		$expirationdate = NULL;
```
Comment (d00p) [inline lib/Froxlor/Api/Commands/Certificates.php]: please lowercase `null` as everywhere in the code

File: lib/Froxlor/Api/Commands/Certificates.php [hunk 2]
```diff
@@ -345,14 +348,16 @@ private function addOrUpdateCertificate($domainid = 0, $ssl_cert_file = '', $ssl
...
+			`ssl_cert_chainfile` = :ssl_cert_chainfile,
+			`expirationdate` = :expirationdate
 			" . $qrywhere . " `domainid`= :domainid
 		");
 		$params = array(
 			"ssl_cert_file" => $ssl_cert_file,
 			"ssl_key_file" => $ssl_key_file,
 			"ssl_ca_file" => $ssl_ca_file,
 			"ssl_cert_chainfile" => $ssl_cert_chainfile,
+			"expirationdate" => date("Y-m-d H:i:s", $expirationdate),
```
Comment (d00p) [inline lib/Froxlor/Api/Commands/Certificates.php]: this will "fail" if `$expirationdate` is **null** - it then outputs `1970-01-01 01:00:00`

Comment (kionez) [reply lib/Froxlor/Api/Commands/Certificates.php]: Ops, you're right, I change the way the variable is assigned to query.

File: lib/Froxlor/Api/Commands/Certificates.php [hunk 3]
```diff
@@ -334,7 +334,7 @@ private function addOrUpdateCertificate($domainid = 0, $ssl_cert_file = '', $ssl
 			} else {
 				\Froxlor\UI\Response::standard_error('sslcertificateinvalidcert', '', true);
 			}
-			$expirationdate = $cert_content['validTo_time_t'];
+			$expirationdate = empty($cert_content['validTo_time_t']) ? null : $date("Y-m-d H:i:s", $cert_content['validTo_time_t']);
```
Comment (d00p) [inline lib/Froxlor/Api/Commands/Certificates.php]:
```diff
- $date(...
+ date(...
```

error in unittests: **Undefined variable: date**

Comment (kionez) [reply lib/Froxlor/Api/Commands/Certificates.php]: Fixed, I'm very sorry... it's better for me to go offline for today :)

Thanks!

Comment (d00p) [reply lib/Froxlor/Api/Commands/Certificates.php]: No need to be sorry, happens :)

**Pull request merged**